src/pkg/egress/syslog/loggregator_emitter.go

```diff
@@ -0,0 +1,40 @@
+package syslog
+
+import (
+	"code.cloudfoundry.org/go-loggregator/v10"
+)
+
+// LogClient is used to emit logs.
+type LogClient interface {
+	EmitLog(message string, opts ...loggregator.EmitLogOption)
+}
+
+type LoggregatorEmitter struct {
+	logClient   LogClient
+	sourceIndex string
+}
+
+// WriteLog writes a message in the application log stream using a LogClient.
+func (appLogEmitter *LoggregatorEmitter) WriteLog(appID string, message string) {
+	if appLogEmitter.logClient == nil || appID == "" {
```

Comment:
I am not sure if the location of this file is sufficient as I am also calling it in the filtered binding fetcher. Should I move this in its separate package? If yes where exactly?

Reply:
Let's leave it as is for now until we think of something better.

Reply:
@chombium it sounds like you're worried that we do not have any mechanisms to prevent future misconfiguration of the `AppLogEmitter` when it's initialized in `syslog_agent.go`, is that right?

I think we could adequately cover this case with a new test in `syslog_agent_test.go`. That would seem like a valuable addition to me.